pkg/imager/iso/uefi.go

```diff
@@ -188,33 +187,30 @@ func CreateUEFI(printf func(string, ...any), options UEFIOptions) error {
 	printf("creating ISO image")
 
 	args := []string{
-		"-as", "mkisofs",
-		"-e", "efiboot.img",
+		"-r",
+		"-J",
+		"-joliet-long",
+		"-e", "--interval:appended_partition_2:all::",
+		"-append_partition", "2", "0xef", efiBootImg,
+		"-partition_cyl_align",
+		"all",
+		"-partition_offset", "16",
+		"-iso_mbr_part_type", "0x83",
 		"-no-emul-boot",
 		"-o", options.OutPath,
 		options.ScratchDir,
-		"--",
-	}
-
-	if epoch, ok, err := utils.SourceDateEpoch(); err != nil {
```

Comment:
`xorriso` tools support `SOURCE_DATE_EPOCH` from env vars and we run imager with it set, so this was redundant. From the man page:

```text
 SOURCE_DATE_EPOCH  belongs  to the specs of reproducible-builds.org.  It is supposed to be either undefined or to contain a decimal number which tells the seconds since january 1st 1970.
       If it contains a number, then it is used as time value to set the default of --modification-date=.  --gpt_disk_guid defaults to "modification-date".  The default of  --set_all_file_dates
       is then "set_to_mtime".  Further the "now" time for ISO nodes without disk source is then set to the SOURCE_DATE_EPOCH value.
       Startup files and program options can override the effect of SOURCE_DATE_EPOCH.
```

I've tested locally setting this produces a reproducible iso (tested with a simple directory since UKI is not reproducible)